### PR TITLE
Rework the remaining inline expectation tests to use the parameterized module

### PR DESCRIPTION
**Pull request by jketema**
Follow up to https://github.com/github/codeql/pull/12789, https://github.com/github/codeql/pull/13346, https://github.com/github/codeql/pull/13426, and https://github.com/github/codeql/pull/13498. This addresses the last remaining Python tests and deprecates the class-based interface.

### CODE REVIEW

File: python/ql/test/experimental/dataflow/TestUtil/FlowTest.qll [hunk 1]
```diff
@@ -37,11 +37,3 @@ private module FlowTest<FlowTestSig Impl> implements TestSig {
...
   }
 }
-
-module MakeFlowTest<FlowTestSig Impl> {
-  import MakeTest<FlowTest<Impl>>
-}
-
-module MakeFlowTest2<FlowTestSig Impl1, FlowTestSig Impl2> {
-  import MakeTest<MergeTests<FlowTest<Impl1>, FlowTest<Impl2>>>
-}
```
Comment (jketema) [inline python/ql/test/experimental/dataflow/TestUtil/FlowTest.qll:47]: I decided to drop this, and just use the bodies directly, as I find these are rather odd looking special cases.

File: python/ql/test/experimental/meta/ConceptsTest.qll [hunk 2]
```diff
@@ -255,52 +255,64 @@ module HttpServerRequestHandlerTest implements TestSig {
   }
 }
 
-class HttpServerHttpResponseTest extends InlineExpectationsTest {
-  File file;
+abstract class DedicatedResponseTest extends string {
```
Comment (jketema) [inline python/ql/test/experimental/meta/ConceptsTest.qll:258]: Note that this moves part of the functionality that first occurred in a number of individual tests to this shared file. I decided not to introduce another parameterized module that then needs to be instantiated, as that would affect many test.

**Review by RasmusWL — DISMISSED** [review]
Thanks so much :heart: :tada: 

I'll hold back from bike-shedding about how to do `DedicatedResponseTest`, since it doesn't really matter :sweat_smile:

**Comment by jketema** [comment]
> I'll hold back from bike-shedding about how to do `DedicatedResponseTest`, since it doesn't really matter 😅

Thanks. I'd be curious to know though how you would have solved this.

**Review by RasmusWL — APPROVED** [review]

**Pull request merged**